cmake/modules/FindMicroHttpd.cmake

```diff
@@ -35,17 +35,7 @@ if(MICROHTTPD_FOUND)
   set(MICROHTTPD_INCLUDE_DIRS ${MICROHTTPD_INCLUDE_DIR})
   set(MICROHTTPD_DEFINITIONS -DHAS_WEB_SERVER=1 -DHAS_WEB_INTERFACE=1)
 
-  if(KODI_DEPENDSBUILD AND NOT WIN32)
-    find_library(GCRYPT_LIBRARY gcrypt)
-    find_library(GPGERROR_LIBRARY gpg-error)
-    list(APPEND MICROHTTPD_LIBRARIES ${GCRYPT_LIBRARY} ${GPGERROR_LIBRARY})
-    mark_as_advanced(GCRYPT_LIBRARY GPGERROR_LIBRARY)
-    if(NOT APPLE AND NOT CORE_SYSTEM_NAME STREQUAL android)
-      list(APPEND MICROHTTPD_LIBRARIES rt)
-    endif()
-  else()
-    list(APPEND MICROHTTPD_LIBRARIES ${PC_MICROHTTPD_STATIC_LIBRARIES})
-  endif()
+  list(APPEND MICROHTTPD_LIBRARIES ${PC_MICROHTTPD_LIBRARIES})
```

Comment:
shouldn't this be PC_MICROHTTPD_STATIC_LIBRARIES?

Reply:
Adding `PC_MICROHTTPD_STATIC_LIBRARIES` unconditionally doesn't work, see PR description.

I thought it is a special pkg-config feature, that for static libs the private libs are also included, but it is caused by the way pkg-config is build in depends (`
--enable-indirect-deps`).
https://github.com/xbmc/xbmc/blob/ce136c8bc4a3a0af3f24752f2a7630d359ae10d6/tools/depends/native/pkg-config-native/Makefile#L19

I'm not sure if we should rely on this. @wsnipex what do you think?

Reply:
probably not, as it will break with static libs outside of our depends.

Reply:
If one uses a toolchain with static libs, shouldn't then pkg-config also be build with `
--enable-indirect-deps`? Otherwise building most of libraries will fail anyway.

Reply:
I don't see why this would be mandatory, so we can't rely on that in general. It's ok for me for depends builds.

We should just check if we found libmicrohttpd.a and only then use  PC_MICROHTTPD_STATIC_LIBRARIES.
Something like https://github.com/xbmc/xbmc/blob/master/cmake/modules/FindFFMPEG.cmake#L182-L190

Reply:
@wsnipex done